### ISSUE TITLE
Feature request, change notification sound

### ISSUE THREAD

**Issue by Luraktinus** [vector-im/riot-web#5891]
Some users (including me) find this high pitched sound unappealing, please let the users change it or go an octave lower.
thank you
--Lue

**Comment by lampholder** [comment]
I wonder - do fb messenger/web whatsapp/slack let you configure notif sounds? Or do they just pick a more universally appealing sound?

**Comment by hdoupe** [comment]
What's the status on this issue? My colleague and I are testing out the app. It's great so far, but this noise gives me a mini-heart attack every time I get a message.

cc @andersonfrailey

**Comment by NintendoManiac64** [comment]
If you're OK with manually doing the process every single time Riot updates, then you can at the very least manually replace the "message.mp3" and "message.ogg" files located in %localappdata%\riot\app-0.17.0\resources\webapp\media

(replace the "0.17.0" with whatever the newest version of Riot is at any given time)

**Comment by itay-grudev** [comment]
Here is a link to the new proposed sound:
https://raw.githubusercontent.com/matrix-org/matrix-react-sdk/f81db65e82873d38ce8fb0a56f7e92b1dd23c7b5/res/media/ring.ogg

**Comment by Maffsie** [comment]
I'm in agreement here, the current notification sound for new messages gives me a not-insignificant amount of anxiety every time I hear it, which means I miss notifications because I need to have the sound turned off. Please give users the option of choosing their own tone without having to reapply a manual file swap on every (frequent!) Riot update.
@itay-grudev Those audio files appear to be for an incoming call, not for a notification of a new instant message.

**Comment by Bandie** [comment]
For Linux users:
You can manually change out/patch the files in `/usr/lib/riot/webapp/media/`

**Comment by t3chguy** [comment]
> /usr/lib/riot/webapp/media/

Sounds like you're using some unofficial distribution, official one stores its files in /opt/riot-desktop/

**Comment by Bandie** [comment]
@t3chguy Well, it seems the package maintainer decided to put things there. 
(I use arch btw. >:) )

**Comment by t3chguy** [comment]
Might want to correct your

> For Linux users:

**Comment by Bandie** [comment]
> Might want to correct your
> 
> > For Linux users:

aight

**Comment by lectrode** [comment]
IMHO, the proposed ringtone is much better than the current. Has a decision been made on whether or not to switch to it?

**Comment by domoritz** [comment]
Besides changing the default, it would be nice to allow people to choose a sound. That was in https://github.com/vector-im/element-web/issues/6776 and merged into this issue.

**Comment by monstro1** [comment]
anyone wanna fix this yet

**Comment by DanH42** [comment]
> I wonder - do fb messenger/web whatsapp/slack let you configure notif sounds? Or do they just pick a more universally appealing sound?

For starters, none of them uses such a **LOUD** notification sound. When my system volume is set such that everything else (notifications, videos, etc) is a comfortable volume, an Element notification is borderline _painful_ it's so loud. I have to keep the tab muted and rely on visual cues to tell when I've gotten a message.

A volume control would be great, but custom notification sounds would also allow for an easy workaround that would also satisfy others who don't like the sound itself, regardless of how loud it is.

**Comment by caniko** [comment]
Why can't we select from a list of ringtones? Why just one default tone?

**Comment by monstro1** [comment]
<h1>when is someone gonna fix this</h1>

i can change the notification sounds for individual rooms, but not the default one. it's unintuitive to have to go into every room and manually replace the sound. atm i've just had to settle with sounds off

**Comment by Igeljaeger** [comment]
welcome to matrix development

**Comment by sl5net** [comment]
here you could here real sound at my android phone. its not edited. its real sound situation https://www.youtube.com/watch?v=OOulxYMrAPQ

**Comment by AverytheFurry** [comment]
Please fucking fix this I have like 5 ppl who have all told me that they don't like using the app because of the notification noise and its really really annoying and literally scares some of my friends lmao like how has this been an issue since 2017 I don't get it this is the most basic feature ever, especially when the default noise is so horrid that it physically causes issues for some users.

**Comment by AverytheFurry** [comment]
> > I wonder - do fb messenger/web whatsapp/slack let you configure notif sounds? Or do they just pick a more universally appealing sound?
> 
> For starters, none of them uses such a **LOUD** notification sound. When my system volume is set such that everything else (notifications, videos, etc) is a comfortable volume, an Element notification is borderline _painful_ it's so loud. I have to keep the tab muted and rely on visual cues to tell when I've gotten a message.
> 
> A volume control would be great, but custom notification sounds would also allow for an easy workaround that would also satisfy others who don't like the sound itself, regardless of how loud it is.

literally this, it physically causes pain like I don't get it please fix it so some people can actually use the app, it literally physically makes some people unable to use it.

**Comment by itay-grudev** [comment]
> Hey @itay-grudev & @elegaanz thanks for this contribution!
> 
> While this may be a marked improvement, we wouldn't like to merge this into the project as is. This clashes with some notification work we're (@matrix-org/design) doing internally and any updates to the sounds would have to be part of a pack with variations for multiple contexts, rather than just one sound being changed arbitrarily.
> 
> We're actually slightly along with choosing some sound packs, but if you'd be interested to talk more feel free to DM me on Matrix: https://matrix.to/#/@nadonomy:matrix.org

How is that work coming along @nadonomy? Maybe consider merging the improved sound from #4500 in the mean time as it has been quite a while now since you posted [that](https://github.com/matrix-org/matrix-react-sdk/pull/4500#issuecomment-620942203).

**Comment by AverytheFurry** [comment]
Sorry if my comment was angry :p I was just upset in the moment, hopefully that doesn't detract from any progress towards this being fixed, sorry.

**Comment by ShadowJonathan** [comment]
> Maybe consider merging the improved sound from #4500 in the mean time

That PR has already been merged a while back, though

**Comment by aaronraimist** [comment]
They meant https://github.com/matrix-org/matrix-react-sdk/pull/4500 presumably

**Comment by c33s** [comment]
this is not only about liking a ringtone or not, the current rington has a physical impact on humans and animals.

to talk via voip i need to set the volume of element to 100% or it is simply to silent to understand the other side. most of the time i forget to reduce the volume after a call to about 15% and the next time someone calls me i nearly get a heart attack. one time the loud noise and the frequency caused a ringing in my ears for a few minutes.

this isn't fun any more. please reduce the volume level in the audio file you are shipping (i can do a PR for that, as i reduced mine by -45db by repacking the asar file) or add a separate volume control for the ring tone and don't wait for the perfect solution with a full blown customized sound-theme.

**Comment by erogol** [comment]
I honestly hate the notification sound. 
It irritates me every time. 
Even if there is a universal sound for every app, I %100 think the element needs a more welcoming sound than the current one.

**Comment by thomasfrobieter** [comment]
> %localappdata%\riot\app-0.17.0\resources\webapp\media

The mp3 files are gone in current versions? So no possibility to replace em manually anymore?

**Comment by Flam3z** [comment]
The ringing sound really hurts my ears, I hope its just a temp change.

**Comment by t3chguy** [comment]
They're in the webapp.asar, you can modify the package using asar tooling.

**Comment by mmance** [comment]
I can't believe people have been listening to that awful ringtone for over 5 years.

**Comment by bb010g** [comment]
What work needs to happen to resolve this issue? Is there a simple patch / set of patches that could be merged to resolve this? Has design figured out what they want notification control to (mostly) work like, but that hasn't been implemented yet? Is this issue completely blocked on design, and these people's problems won't be fixed until design figures out new notification control and that's implemented?

It's not clear to me where this issue really is right now.

**Comment by kellyclowers** [comment]
Please fix this, the current sound is unbearable. Change it something less horrible, let us select from a few, let us easily replace the file... something!

**Comment by numanair** [comment]
I have stopped using Element because of this issue. It is literally unbearable.

**Comment by ShadowJonathan** [comment]
I understand that for Element it is hard to change - what is essentially - part of their brand.

However, at the very least the interface for someone to change their notification sound should be exposed, https://github.com/vector-im/element-web/issues/20286 is this feature request, and this feature is already in the code, as detailed [here](https://github.com/vector-im/element-web/issues/2105#issuecomment-998217391), the only thing missing is UI that would change that settings value.

**Comment by mibmo** [comment]
> @bb010g I made a PR a while ago. They just didn't want it, saying they were working on something of their own. They just never released it in the last several years.

Could you perhaps link your PR? We might be able to bump it.

**Comment by aselus-hub** [comment]
just want to chime in that this is bad, I know a lot of people that outright don't want to use element or just silence it so it's impossible to use due to the sound it makes, which is a prime example of how this hurts adoption. Can we please get some official word on this and the proposed "other change"?

**Comment by NukeManDan** [comment]
> At the very least the interface for someone to change their notification sound should be exposed, #20286 is this feature request, and this feature is already in the code, as detailed [here](https://github.com/vector-im/element-web/issues/2105#issuecomment-998217391), the only thing missing is UI that would change that settings value.

is it possible without building source locally to use this via, say a cli command? Really anything at his point so I can keep some sound on.

**Comment by mibmo** [comment]
If all that's missing is the UI, would anyone be willing to pick this up and churn through it? Seems (almost deceptively) simple. Just a few file pickers?

**Comment by tristan-cottam** [comment]
Having an option to change the ringing sound, despite being optimal, is less important than changing the default sound itself. Many users would be turned off by hearing this sound in the first place, some of them dropping the app altogether without bothering to look for such a setting, let alone expecting such a setting to even exist.

Even a basic looping beep would suffice, and would furthermore probably be the most universally user-friendly solution.

**First impressions outweigh customizability.**

**Comment by mibmo** [comment]
For reference, [these instructions](https://github.com/vector-im/element-web/issues/2105#issuecomment-998217391) currently work, but it's not exactly ideal.

> @tristan-cottam: First impressions outweigh customizability.

Not a bad take. Do you have anything specific in mind? I think I'll pick this issue up, starting with customisation.

**Comment by tristan-cottam** [comment]
> Even a basic looping beep would suffice, and would furthermore probably be the most universally user-friendly solution.

If this not what you were asking for, please detail your expectancies.

**Comment by mibmo** [comment]
Woops, yes! _note to self: don't work on OSS in the middle of the night!_
Are you referring to the ringing or notification sound though? what you're describing definitely wouldn't work for notifications (at least the way I'm imagining it).
Nonetheless if you can produce it yourself then an audio file (with accompanying compatible license) would be very nice :)

**Comment by Alkalurops** [comment]
How is this not fixed. I'm going to stop using this. It's unbearable with the sound and useless without a notification.

**Comment by viasux** [comment]
> How is this not fixed. I'm going to stop using this. It's unbearable with the sound and useless without a notification.

yeah I exclusively use signal for voice calls now, it was too horrendous to deal with..

alternatively, cinny.in is a good text only matrix client

**Comment by ThiefMaster** [comment]
This is an absolute must have, with separate sounds for message and calls (ideally separate choices for all different sounds!).

My 80-year-old dad is using Element on his PC, smartphone and tablet. He does NOT always carry the phone with him at home (and good luck convincing someone of that age to change their habits). So when there's the short windows notification sound and he's in the room next to the one with his PC or watching TV in the same room (somewhat loudly since his ears aren't the best anymore), he does NOT notice the notification sound and ends up missing the message until he sees it on the PC.

With configurable sounds he would simply pick a longer one and even though personally I would find that super annoying for new incoming messages, he'd certainly notice it when it's not just such a short noise.

TL;DR: Please let people choose their sounds.

**Comment by ashledombos** [comment]
Here is an issue that has been running for 5 years, that could be easily solved, or at least have a simple workaround, (it seems much simpler than dealing with forked threads, for instance) but yet ignored and causing a hemorrhage of users… Why ?

**Comment by Ryonez** [comment]
Seems like the devtools method no longer works.

**Comment by autoteelar** [comment]
> Here is an issue that has been running for 5 years, that could be easily solved, or at least have a simple workaround, (it seems much simpler than dealing with forked threads, for instance) but yet ignored and causing a hemorrhage of users… Why ?

yeah genuinely, schildichat already added a custom noise feature because the default one was just too ear piercingly loud and annoying...

i think its about time element did too. 

the fact this has been ignored for 5 years, does not look good on the element devs in relation to the community. if lots of people want something that is genuinely beneficial, there is no reason to not do it, this would take like 5 minutes to code tops.

**Comment by MurzNN** [comment]
@kittykat Can you please recheck if this feature is highlighted in Roadmap here https://github.com/vector-im/roadmap/projects/1 ? If not, could you please add and arrange it, to not lost this crucial feature?

**Comment by jimkoen** [comment]
> This is still happening, only it's part of the design team's (large) workload

At what point are we allowed to make snarky comments about what seems to be willful ignorance?

> I understand that for Element it is hard to change - what is essentially - part of their brand.

It's okay, other brands have made workplace anxiety part of their brand culture as well!

> some friends have noted that the sound is annoying, gives anxiety, or is downright unbearable on their ears.

I am literally waiting for that one GTK based project to reach a usable state on Mac OSX, then I'm ditching element forever.
It's absolutely beyond me why you'd sacrifice a user bade over some atrocious, and quite honestly barely noticeable brand design.

This isn't aimed at you personally ShadowJonathan and more meant as a dig towards the eternally busy design team.

**Comment by asdil12** [comment]
As the package maintainer of the openSUSE rpm package of element I wonder if I could make it possible to let users replace the audio file (using a symlink - maybe via update-alternatives?).

The question is if that file is within the asar bundle or outside of it.

**Comment by autoteelar** [comment]
> > This is still happening, only it's part of the design team's (large) workload
> 
> At what point are we allowed to make snarky comments about what seems to be willful ignorance?
> 
> > I understand that for Element it is hard to change - what is essentially - part of their brand.
> 
> It's okay, other brands have made workplace anxiety part of their brand culture as well!
> 
> > some friends have noted that the sound is annoying, gives anxiety, or is downright unbearable on their ears.
> 
> I am literally waiting for that one GTK based project to reach a usable state on Mac OSX, then I'm ditching element forever. It's absolutely beyond me why you'd sacrifice a user bade over some atrocious, and quite honestly barely noticeable brand design.
> 
> This isn't aimed at you personally ShadowJonathan and more meant as a dig towards the eternally busy design team.

i can give you some workarounds if you want. 

1. compiling it yourself but replacin the files before compiling
2. use schildichat, its element but the desktop app is way better basically
3. idk about that one GTK one, if you could share im mildly interested, there is a QT one called Nheko but i didnt really like it either.

if you host your own element web the mp3 files for notifications can be changed pretty much any time yourself

**Comment by autoteelar** [comment]
> As the package maintainer of the openSUSE rpm package of element I wonder if I could make it possible to let users replace the audio file (using a symlink - maybe via update-alternatives?).
> 
> The question is if that file is within the asar bundle or outside of it.

from the looks of it its literally just an mp3 file, you could just allow the user to change that file via gui, its extremely simple to do

**Comment by autoteelar** [comment]
maybe something like this? 
![image](https://user-images.githubusercontent.com/76745649/198837010-03d77547-6b6a-4c77-a8d0-d690ee027868.png)

**Comment by autoteelar** [comment]
this guys screenshot example uses wav, but personally i would use mp3 or flac

**Comment by NintendoManiac64** [comment]
> this guys screenshot example uses wav, but personally i would use mp3 or flac

Or OGG vorbis or Opus...

Or really just anything that Chromium supports via HTML5 (including M4A／AAC).

**Comment by autoteelar** [comment]
yeah those 3 are also really good, i think it should support as many file types as possible.

**Comment by ShadowJonathan** [comment]
I encountered this issue again with some friends signing up for matrix through element getting met by this notification sound.

Design and Product, how it is YET STILL that this issue is unaddressed, for multiple years, while by your own prioritisation rules you should have at least dropped other projects by now because this issue has `O-Occasional` *and* `S-Major` on it?

I want to once again remind y'all of https://github.com/vector-im/element-web/issues/20286, which is an easy-enough stop-gap solution to at least allow users to change this without having to change local files, which some of those friends had to do, out of necessity, as the notification sound was extremely distressing to them.

**Comment by Alkalurops** [comment]
I deactivated all sounds for Element and only see new messages arriving once in a while when I check for other messages incoming (Threema, iMessage, …).

**Comment by viasux** [comment]
> I encountered this issue again with some friends signing up for matrix through element getting met by this notification sound.
> 
> Design and Product, how it is YET STILL that this issue is unaddressed, for multiple years, while by your own prioritisation rules you should have at least dropped other projects by now because this issue has `O-Occasional` _and_ `S-Major` on it?
> 
> I want to once again remind y'all of #20286, which is an easy-enough stop-gap solution to at least allow users to change this without having to change local files, which some of those friends had to do, out of necessity, as the notification sound was extremely distressing to them.

honestly just use https://github.com/cinnyapp/cinny/

**Comment by Ryonez** [comment]
> honestly just use [cinnyapp/cinny](https://github.com/cinnyapp/cinny/)

Sadly it doesn't have feature parity with Element. No threads is one example.

**Comment by vertigo220** [comment]
I have to agree it's pretty insane this issue has been outstanding for this long, considering how simple it is and that Matrix/Element are trying to become more widespread yet this is a fairly big barrier to that, again one that can very easily and quickly be fixed. It's been almost two years since a new sound was proposed, and all it would take would be for the default sound file to be replaced by the new one in the build, yet even that's not been done. What possible, valid reason could there be for this? Now I have to go and do it manually, something a user shouldn't have to do, and most users won't want or even know how to do, because it feels like it's going to give me a heart attack every time I get a call.

Also, this is a duplicate of #5031, and at least some of those that gave that issue a thumbs up didn't do so here, which means there's even more in favor of this than reflected here, and that's not to mention all those that have likely tried Element and trashed it due to this without even bothering to come make an issue, as I think most users of programs don't participate on GitHub.

**Comment by ShadowJonathan** [comment]
FWIW this issue is about the notification sound *and* the ringing sound, i dont think it should be dedupped into #5031 for that reason, as I'm primarily agitated by the notification sound

**Comment by vertigo220** [comment]
Fair enough, and not necessarily saying it should be, though it could be if that one were expanded to involve both. Bottom line is they both deal with essentially the same issue, which is that the sounds in general, both ringtone and notifications, are not very good, as evidenced by the significant number of people participating in these issues and #2105, and they need to at least be changed to something better and ideally allow the user to change them to whatever they want. I actually think if anything that issue, despite being first, should be merged into this one, since this one is much more active in both discussion and thumbs up. But I really mentioned it not to say it was a duplicate so much as to say this problem is fragmented and, as such, has more support for a resolution than any one issue would suggest.

**Comment by amunrarara** [comment]
Honestly, the _less stressful ringtone_ proposed still gives me a heart attack when it rings, particularly when I'm wearing headphones. It honestly couldn't be _that_ difficult to add a GUI option. 

Anyone interested in opening their own PR for this?

**Comment by ShadowJonathan** [comment]
@amunrarara https://github.com/vector-im/element-web/issues/20286

**Comment by kenwuuu** [comment]
I'm working on adding the ability for users to change their message notification sound, it will sync across devices. Not working on VOIP ringtone. Will provide updates in the next couple days, just wrapping up.

**Comment by amunrarara** [comment]
Great, thanks for your work!

**Comment by tristan-cottam** [comment]
@nadonomy Any updates on that notification redesign? Is it still ongoing?

It's been just under 3 years since [your last reply on the subject](https://github.com/matrix-org/matrix-react-sdk/pull/4500#issuecomment-622001561); If you could just give us a sign that you're still alive and well, and let us know what's going on, it would be much appreciated.

**Comment by eslerm** [comment]
Please updates the tags of this issue.

[Less stressful ringtone #5031](https://github.com/vector-im/element-web/issues/5031) was closed as a duplicate to this issue. #5031 is a `O-Frequent` accessibility (`A11Y`) issue.

Please update the tags to include `O-Frequent` and `A11y`.

**Comment by robintown** [comment]
Thanks for raising, @eslerm, I've reopened https://github.com/vector-im/element-web/issues/5031 since we consider it important that the product use a more pleasant default ringtone, regardless of the outcome of the feature request tracked in this issue.

**Comment by KaKi87** [comment]
Here's an urgent implementation proposal : make the app look for an eventual `message.mp3`/`message.ogg` file at `~/.config/riot`/`%userprofile%/riot` before loading the default one.

This should be doable in a matter of minutes, requires zero GUI changes and solve the overwrite on upgrade issue.

Thanks

**Comment by t3chguy** [comment]
> This should be doable in a matter of minutes, requires zero GUI changes and solves the overwrite on upgrade issue.

You're more than welcome to do it in that case, you make it sound so easy.

**Comment by KaKi87** [comment]
Well, it should be drastically easier than creating a ringtones CRUD database and GUI, that's probably why such a thing still hasn't been implemented, which I understand, but at the risk of repeating others, this issue really has unbearable consequences for impacted users, which is why I'm making this suggestion for a temporary solution with hope to see it happen as soon as possible.

**Comment by KaKi87** [comment]
Oh, the file is accessed directly from the renderer process... Well, setting up a CRUD will require serving the file from the main process anyway.

**Comment by piegamesde** [comment]
@KaKi87 if you urgently want a temporary solution and are hosting your own Element Web instances, simply patching it can be as short as a handful of lines: https://git.maralorn.de/nixos-config/tree/overlays/element-web/default.nix

Personally, I don't want any temporary hacks, I'd like to see a proper implementation of the feature instead.

**Comment by t3chguy** [comment]
> Well, setting up a CRUD will require serving the file from the main process anyway.

Nope, it'd be stored in Matrix, so its accessible from all your clients. Just like all user-configurable avatars are.

**Comment by t3chguy** [comment]
> That's overkill, would take even more time, and will disadvantage users of outdated homeservers.

If your homeserver is so outdated that it doesn't support MXC then you won't be able to use it with any Element version from the past 3 years. No special support is needed beyond Matrix Media Repository (MXC) which already backs avatars and in-chat media.

Its not overkill, its exactly how the Room Settings > Notifications sound override already functions today, so arguably copy and pasting that code is the opposite of overkill.

**Comment by ThiefMaster** [comment]
I would NOT want to be forced to use the same ringtone on all devices! So storing the file in Matrix cool, always syncing the choice not cool.

**Comment by t3chguy** [comment]
> So, this endpoint allows uploading arbitrary files without being an avatar nor a message attachment ?

Yes, in an encrypted messenger the server has no way of telling what you upload, nor can it see references to it, because its encrypted.

> And even so, the file would have to have a unique name which would have to be stored somewhere server-side that doesn't exist either.

Sure it does. account_data.

> Unless there's some kind of key-value store available as well ?

I suggest playing with `/devtools` one of these days in Element Web/Desktop, everything is extensible.

> I would NOT want to be forced to use the same ringtone on all devices! So storing the file in Matrix cool, always syncing the choice not cool.

Whether per-device granularity is enabled for this would be up to the product team

**Comment by KaKi87** [comment]
Alright, considering the back-end's extensibility, server-side ringtone customization sounds nice.

But, if that option is chosen, will it be implemented anytime soon ?

**Comment by jellykells** [comment]
>will it be implemented anytime soon ?

no it's been six years lmao

**Comment by KaKi87** [comment]
See, that's why I'm making this urgent implementation proposal instead.

**Comment by t3chguy** [comment]
The technical implementation side of this isn't the blocker, the product/design side is, we can't land it without sign off from them.

**Comment by KaKi87** [comment]
What's that ?

**Comment by t3chguy** [comment]
The product team, you asked who's in charge

> Who's in charge then ?

> The technical implementation side of this isn't the blocker, the product/design side is, we can't land it without sign off from them.

The product team are.

**Comment by KaKi87** [comment]
I mean where should I copy/paste "@vector-im/product" ? I tried to find a GitHub repo at that location but there isn't. So, where is it ? On Matrix or something ?

**Comment by t3chguy** [comment]
It is a Github team

**Comment by KaKi87** [comment]
Where's that screenshot from ?

I tried [the *People* page](https://github.com/orgs/vector-im/people) but it only returns ungrouped individuals.

**Comment by t3chguy** [comment]
I'm fairly certain the underlying members are kept private for privacy, but you can ping the team as you did above already.

**Comment by KaKi87** [comment]
What ? I never pinged anyone in this issue.

**Comment by t3chguy** [comment]
<img width="817" alt="image" src="https://github.com/vector-im/element-web/assets/2403652/a861dffe-7726-4255-915a-7311b4abbadf">

**Comment by KaKi87** [comment]
Oh, but since the beginning I'm seeing plain text, how are you seeing a hyperlink ?

**Comment by t3chguy** [comment]
My guess is differing github permissions

**Comment by piegamesde** [comment]
I have total understanding for not being able to customize the notification sound, for the reasons outlined.

However, why the default sound has not been changed to something more pleasant is beyond my understanding. (Tbf this is tracked in a separate issue, which should be bumped in priority IMO.)

Have a look at the other (desktop) messengers. Do users come in crying that they want to customize the notification sound? As far as I know, no. Why? Because theirs does not suck. I think it's fair to say that most users don't want a *custom* notification sound, they just want a *good* one.

**Comment by ShadowJonathan** [comment]
Alright, I have been understanding of Element's priorities for at least a long while now, but this has crossed my personal threshold for how much patience I have for this kind of corporate behaviour.

Since this is clearly not a priority, and still affecting a lot of users, either Element employees have gotten deaf or muted the sound themselves, or they simply don't care about accessibility like this, since there is clearly a breakdown of communication between those who prioritise, and those who consider user accessibility like this.

Therefore, I don't think Element cares about user accessibility, in the end, at all, beyond what looks good and what's flashy, if this is *still* on the backburner, compared to other features and fixed being shipped left and right.

I understand Element's financial situation, but I have lost enough patience to care, and am now seeing this as a deficiency of theirs to not come back to the community on this.

**Comment by t3chguy** [comment]
> either Element employees have gotten deaf or muted the sound themselves, or they simply don't care about accessibility like this

@ShadowJonathan you are aware https://github.com/matrix-org/matrix-react-sdk/pull/9143 landed, yes? It'll be in today's release. Its not the final state of the situation but it resolves the primary complaint of volume.

> I don't think that, in the end, Element cares about user accessibility

As someone who has personally spent many weeks both paid and unpaid working on accessibility you can rightly fact check yourself.

**Comment by grxnola** [comment]
If you've needed the feature for several years, you could always submit a PR

**Comment by piegamesde** [comment]
@grxnola You mean like this? https://github.com/matrix-org/matrix-react-sdk/pull/11078

**Comment by lectrode** [comment]
> @ShadowJonathan you are aware [matrix-org/matrix-react-sdk#9143](https://github.com/matrix-org/matrix-react-sdk/pull/9143) landed, yes? It'll be in today's release. Its not the final state of the situation but it resolves the primary complaint of volume.

While that is certainly a small step in the right direction (it took nearly an entire year just to accept a PR to *reduce the volume* of some of the sounds?! Wtf...), that does not resolve the primary complaint that the existing ring/notification sounds are jarring enough to wake the dead.

> As someone who has personally spent many weeks both paid and unpaid working on accessibility you can rightly fact check yourself.

I don't think anyone is questioning work ethic of contributors - only the apparent strictness with which PRs and other improvements are adopted. A review and approval process is definitely a good thing, but there is such a thing as "letting the perfect be the enemy of the good". In this case, smaller improvements that provide a net benefit are rejected because there are projects "in the works" that will resolve something related.

That would be fine, except that those projects take years to complete (the supposed sound theming project has been in progress for over 7 years), leaving a  gaping accessibility hole in the client for everyone during the interim. The proper solution for this would be to adopt a *better* ring and notification sound for now, and provide a refined/"perfect" sound theming solution later.

**Comment by epicEaston197** [comment]
Bro when I first heard the notification sound for ringing I thought my spaceship was under attack

**Comment by luixxiul** [comment]
This is a cross-post from https://github.com/vector-im/element-web/issues/5031#issuecomment-1663412472 to let you guys know that [schildichat](https://schildi.chat/desktop/), an Element-based client, has implemented an option to select a less stressful notification sound which should not be so jarring. 

![Screenshot_2023-08-03_07-08-45](https://github.com/vector-im/element-web/assets/3362943/af7e2bc2-bcbc-4b77-81d1-690e41af146b)

> The proper solution for this would be to adopt a _better_ ring and notification sound for now, and provide a refined/"perfect" sound theming solution later.

**Comment by adamency** [comment]
@daniellekirkwood
> We are not looking at allowing folks the ability to personalise the sounds of Element as a whole.

Please explain the reasoning behind this decision ? You have simply stated this without ever explaining why this idea was rejected. I don't understand how changing a sound file can be problematic.

**Comment by cknoll** [comment]
As I understand this issue will remain open quite a while. While I personally do not like the element sounds I can live with this becaus as a whole I really like the app.

Anyway it would be nice to have some collections of workarround in this issue. Some of the first messages discuss which files to change manually. I would like to do so but I use element via **flatpak**. If anyone knows how to exchange the media files inside a flatpak container, please let me know.

**Comment by daniellekirkwood** [comment]
> @daniellekirkwood
> 
> > We are not looking at allowing folks the ability to personalise the sounds of Element as a whole.
> 
> Please explain the reasoning behind this decision ? You have simply stated this without ever explaining why this idea was rejected. I don't understand how changing a sound file **on your own instance** can be problematic.

We will be reviewing sound as a holistic topic across our product set in the future. Until then, we are focussing our efforts on gnarly bugs, things such as stuck notifications and threads issues. 

When we get to reviewing Sound and our notifications interactions as a whole we'll aboslutely consult folks here and take all of this feedback into consideration. 

Please bear in mind: There are a lot of switches and customisations in our product and this makes Element harder to use for many people. We are trying to reduce the number of customisations in the product and re-introduce them purposefully with the right level of consideration and feedback.

**Comment by cknoll** [comment]
> We will be reviewing sound as a holistic topic across our product set in the future.

I, personally understand this approach. After all I use element as a free product and I do not have any right to arrogate specific features or to decide how to prioritize development.


On the other hand the developers should keep in mind that sound is an emotional topic and sounds which are perceived as "bad" can alienate users much more than some (minor) bugs. I think the general problem is that people in this issue are probably not affected by other bugs and issues and see the sound problem as the one big thing. Therefore, they lack understanding why this takes so long (issue open since 2017).


I think, compiling some comunity-contributed tutorials for workarounds (like manually changing the sound files) would help here. Because people who are really passionate about this topic then would have another option then just wo wait (or abandon element).

**Comment by adamency** [comment]
> Please bear in mind: There are a lot of switches and customisations in our product and this makes Element harder to use for many people. We are trying to reduce the number of customisations in the product and re-introduce them purposefully with the right level of consideration and feedback.

I can understand this. However, couldn't we set a default and then simply allow in settings to modify the sound if wanted ? This wouldn't add another **necessary** configuration option and still allow folks to change the sounds if they want to.

**Comment by Californian** [comment]
> > @daniellekirkwood
> Please bear in mind: There are a lot of switches and customisations in our product and this makes Element harder to use for many people. We are trying to reduce the number of customisations in the product and re-introduce them purposefully with the right level of consideration and feedback.

I think this is the main point of tension, but I totally understand where you're coming from from a product perspective: 90%+ of users will never change a default notification sound, but all users might see the clutter that would come from adding a UI for the option!

"Customization reduction" is, however, is a terrifying thing for any power user to ponder and, depending on implementation, risks significant user churn among the most active users and advocators of the product.

There is, however, a very common UX pattern for dealing with this tension that I sincerely hope you're adopting (for this issue and in general): an "advanced"/"developer" mode. Let power users tinker and change stuff in a maximalist settings screen world, let normal users get their messages reliably without clutter. It's really the best of both worlds! It's already in use in some sections of the UI, but a global toggle (or settings-tab-level toggles, but not the current intra-tab, section-level toggles) would do wonders to reduce cluter while still allowing for the flexibility of contributions like the PRs enabling users to choose a different sound.

Just my 2 cents!

**Comment by ThiefMaster** [comment]
> "Customization reduction" is, however, is a terrifying thing for any power user to ponder and, depending on implementation, risks significant user churn among the most active users and advocators of the product.

Yes. I had to restrain myself not to write a much more strongly-worded comment as a reply to this quote (which I hadn't seen before). Because I'm fucking sick of applications dumbing down their stuff all the time, and every time they do this I tend to use some feature/switch which I found very useful.

Just have a simple set of basic settings for "normal users". Good defaults, and easy to change those where the defaults may not be everyone's liking (ctrl+senter vs enter to send for example - all the messengers like Skype that removed this added it back ;)) 

And then as @Californian suggested hide the rest under some "advanced settings" toggle. Maybe not as crazy as `about:config` in Firefox, but just so power users can still customize things...

**Comment by KaKi87** [comment]
Concretely : what in the world do we have to do to get this feature now ?!

**Comment by eslerm** [comment]
> Concretely : what in the world do we have to do to get this feature now ?!

Community requests for accessibility will likely continue not being a priority https://github.com/vector-im/element-web/issues/5031.

If you work for an organization or government which is considering hiring or granting money to Element.io or Matrix developers, make sure their sales reps know that accessibility issues, like this, are important.

**Comment by Bugsbane** [comment]
Take a listen to the ringtone here:
https://freesound.org/people/Thoribass/sounds/254822/

The player has a "loop" option, just click the refresh-looking, two arrows in a circle icon.

I think most people would prefer this to the current sound, and it's already permissively licensed (CC-BY). It's certainly much less stress-inducing!

**Comment by ShadowJonathan** [comment]
Happy belated 6-year issue birthday, by the way.

**Comment by VermiumSifell** [comment]
I really think this feature should be implemented, will probably make more power users ask more friends to join the Matrix world too.

**Comment by sl5net** [comment]
> 90%+ of users will never change a default

Having separate simple and advanced options menus is a great way to cater to both beginner and experienced users in your app. 

**Simple Menu:**

* Focus on the most essential functionalities. 

**Advanced Menu:**

* Offer more granular control. This could include different sounds.

**Comment by autoteelar** [comment]
i havent used element in years, schildichat lets you change it

On Friday, May 10, 2024, sl5net ***@***.***> wrote:
> 90%+ of users will never change a default
>
> Having separate simple and advanced options menus is a great way to cater
to both beginner and experienced users in your app.
>
> Simple Menu:
>
> Focus on the most essential functionalities.
>
> Advanced Menu:
>
> Offer more granular control. This could include different sounds.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.<
https://ci3.googleusercontent.com/meips/ADKq_NZ_oaBABjfii55_MZH2kpHMMcj-HiRk5klKj4gyEhNpiqSoJRNoSwrQzZ-d97h_iIyKU5O5R-h6ps3BpfIqLaTk6CwiA0EAMbbZQZV5wNXMOSVIlLUsSjyiGY9Eyx8mMcUaa5P3OgIJicQgB66QixTmbwt03pimLAsqoqXZ6-0HzbF9i-IjYeM35H7JMO2dMqyi7UN9rwdG00CS2Z7W7wM3382aep6fyxcQw-HZaGULWH8=s0-d-e1-ft#https://github.com/notifications/beacon/ASJQXMNDMZH7SXTIQWDP63DZBTUGLA5CNFSM4EJXUIS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOPV2U7PQ.gif>Message
ID: ***@***.***>

**Comment by sircoderin** [comment]
Can we please have an update on this? Me and my friends literally avoid using Element because of the horrifying ring tone.

**Comment by noideastowork** [comment]
Where can I find the notification sounds on macOS so that I can replace them myself?

**Comment by 0xilly** [comment]
Is there any update on this or could the element come out with an official statement that they're not working on fixing this and close the issue so we can move on.

**Comment by RubensRainelli** [comment]
> I wonder - do fb messenger/web whatsapp/slack let you configure notif sounds? Or do they just pick a more universally appealing sound?

They use a more appealing one.

The actual notification sound of Element is quite similar to System Error in Linux or Windows or Mac, I don't remember which one.

There should be even more options like multiple different sounds, like:

- reply or direct message and tagging
- @room messages and multiuser messages.

**Comment by theGiallo** [comment]
Oh, wow. This is old. Thanks @t3chguy , sorry for the duplicate. I didn't fint this post when I searched.

**Comment by ghost** [comment]
> The actual notification sound of Element is quite similar to System Error in Linux or Windows or Mac, I don't remember which one.
> 


Nautilus and Terminal error in Ubuntu is really similar, if memory serves.

Edit: apparently I can set a custom notification sound per room!  Not sure if this works outof the box (not admin)

**Comment by Bugsbane** [comment]
Being able to set the sound by room helps, but I'm not going to be able to get my aging, tech-phobic family members to go through and set custom ring tones for each other family member they talk to in Element.

First: the default should just be better. I've already posted a freely (CC0) licensed alternative that I'm sure most people would agree is vastly superior. I've linked to the file. It's literally just waiting to be uploaded and this issue is largely fixed.

Second: it should be easy for anyone to just set a default ring tone that *every* caller triggers. It shouldn't depend on going through room by room, having to update each.

**Comment by KaKi87** [comment]
This really makes me wonder how the maintainers live with that ringtone.

**Comment by boreq** [comment]
My eardrums were just almost blown by element playing a notification sound about 10000000 dB louder than my music. I muted all notifications as a fix but this must be some kind of a bug, no other firefox tab makes sounds this loud. I don't know what's happening here but it would be cool if this didn't happen again.

**Comment by epicEaston197** [comment]
> My eardrums were just almost blown by element playing a notification sound about 10000000 dB louder than my music. I muted all notifications as a fix but this must be some kind of a bug, no other firefox tab makes sounds this loud. I don't know what's happening here but it would be cool if this didn't happen again.

That sucks that sounds like an issue you need to report separately from this

**Comment by boreq** [comment]
> That sucks that sounds like an issue you need to report separately from this

It may just be the same issue and it may just be about how sharp that sound is. It actually scared me and made me jump in my chair. Not the first time this happened either.

**Comment by foresto** [comment]
> It may just be the same issue and it may just be about how sharp that sound is.

This is likely. The default message sound has always been unforgivably, unforgivingly harsh. The amplitude was reduced somewhat in d381fa3f2c7572d8a28814955e8f74efd4b13e3e, but even now that it's slightly quieter than it was until 2023, it remains quite jarring when played through half-decent loudspeakers above minimal volume.

Whenever I'm forced to use Element, one of the first things I do is upload a different message sound. (This is a painstaking process, since as far as I know Element still requires that it be done individually for each room, but I value my hearing and sanity enough to justify the effort.) I usually pick a sound from a desktop sound pack or some other web-based messaging service, and every single one that I've ever used was fine with no changes to my system's volume. Element's message sound is exceptionally hostile.

(Side note: Anyone here seeking an alternative message sound might try a web search for "creative commons pop sound". For example, [this one](https://pixabay.com/sound-effects/pop-402324/) took less than a minute to find, and there are surely more out there.)

**Comment by ahmeedev** [comment]
Is there any update on this feature?

**Comment by t3chguy** [comment]
For now design has chosen a new sound https://github.com/element-hq/element-web/issues/30760 - it will replace the current sound but no option will yet be available to provide your own

**Comment by ShadowJonathan** [comment]
#30804 has been merged 👀

**Comment by rkrisztian** [comment]
**My relatives can hardly hear the new ringtone.** For us this new version (v1.12.1) is even worse because of that. Customization is the way to go. Sorry, but people in https://github.com/element-hq/element-web/issues/30760 are right to complain too.

For now I did this as a hot-patch workaround (not perfect: have to be re-applied for each new release), which I'm sharing below:

1. Find location of `webapp.asar`, e.g., under Linux Mint using Flatpak: `/var/lib/flatpak/app/im.riot.Riot/current/active/files/Element/resources/webapp.asar`
2. Copy it to a temporary directory where you work on that archive, e.g., `mkdir ~/tmp/element-hotpatch; cd ~/tmp/element-hotpatch; cp /var/lib/flatpak/app/im.riot.Riot/current/active/files/Element/resources/webapp.asar ./`
3. Create a directory to unpack the archive file into: `mkdir webapp`
4. Install node/npm somehow, doesn't matter how, e.g., use NVS (Node Version Switcher), `echo '22.17.0' > .node_version; nvs use`
5. Install the asar packer/unpacker tool: `npm install @electron/asar`
6. Unpack the archive into the `webapp` folder: `npx asar extract webapp.asar webapp/`
7. Locate the ringtone files: `(cd webapp; find . -name ring.mp3; find . -name ring.ogg)`, should be: `/media/ring.mp3` and `/media/ring.ogg`.
8. Replace the MP3 with whatever you want.
9. Convert the MP3 to OGG: `(cd webapp/media; ffmpeg -i ring.mp3 -codec:a libvorbis -qscale:a 8 ring.ogg)`
10. Re-pack the archive: `mv webapp.asar webapp-orig.asar; npx asar pack webapp webapp.asar`
11. Copy the archive back to its original location: `sudo cp -f webapp.asar /var/lib/flatpak/app/im.riot.Riot/current/active/files/Element/resources/webapp.asar; sudo chown root: /var/lib/flatpak/app/im.riot.Riot/current/active/files/Element/resources/webapp.asar; sudo chmod 644 /var/lib/flatpak/app/im.riot.Riot/current/active/files/Element/resources/webapp.asar`

**Comment by Taranchul** [comment]
> **My relatives can hardly hear the new ringtone.** For us this new version (v1.12.1) is even worse because of that. 

I also don't like the new sound. It's barely noticeable if any other application outputs sound (e.g. playing a video or music, being in a voice chat etc.) or if you're in a noisy environment. It's more of a murmur rather than a notification.

**Comment by Taranchul** [comment]
By the way, you can find the old sounds in their history:

https://github.com/element-hq/element-web/commits/develop/res/media/message.ogg

**Comment by Athwale** [comment]
I would be really grateful if I could choose my own notification sound. Linux provides a plethora of built in sounds, would it be possible to let us pick system sounds that I can replace or add to if I want?

**Comment by t3chguy** [comment]
@Athwale looks like only Electron on macOS supports using a chosen system sound https://www.electronjs.org/docs/latest/api/notification#playing-sounds otherwise it's Chrome's default sound or the applications own chosen sound

**Comment by joelmichael** [comment]
I found the old sound more pleasant. I miss it. I have fond memories associated with it.

**Comment by ilu33** [comment]
Changing something so individual like a sound to another one just because several people complained was a bad idea. At least without offering a rollback for the vast majority that had no reason to complain about the old sound. Or a customization.

The new sound rendered element web unusable. I'll switch off updates for element desktop for now until there is a way to revert the change.

**Comment by mrkesu** [comment]
So I see https://github.com/element-hq/element-desktop/issues/2296 was closed as a duplicate for this even though it was the desktop application. I guess the solution was to just change it to another sound people dislike? Well, I mean it's something.

Anyways; It is not at all unreasonable to let users choose their own notification sound. One might even say this is the norm in most modern chat apps.

The notification sound is awful _- at least in the desktop app -_ and I just turned it off completely rather than listen to that.

**Comment by ThiefMaster** [comment]
Just to add some context: My 84-year-old dad is using Element. As you can imagine, people's hearing at this age is not as good as it's for those of us who are still young.

While the old sounds were absolutely awful (especially the sound when making an OUTGOING call), they were very noticeable and easy to hear, even from the next room.

The new notification sound is way too soft and he literally told me that often he doesn't notice it.

So just do the right thing and let people select some local sound file in the client. No need to sync it. That would be a nice to have *option* later on, but just the ability to use a different sound w/o losing the change after every single update (due to the messy way of how electron apps are installed/updated) would already be great.

**Comment by t3chguy** [comment]
> So I see https://github.com/element-hq/element-desktop/issues/2296 was closed as a duplicate for this even though it was the desktop application.

Element Desktop = Element Web + Electron + Seshat, so yes Element Desktop issues which also apply to web are closed in favour of web ones.

> So just do the right thing and let people select some local sound file in the client. No need to sync it. 

It is already possible in Room Settings > Notifications - contributions welcome for global options. https://github.com/matrix-org/matrix-react-sdk/pull/10494 was an attempt at this but didn't end up in a mergeable state as per https://github.com/matrix-org/matrix-react-sdk/pull/10494#issuecomment-2139164358

**Comment by mrkesu** [comment]
> It is already possible in Room Settings > Notifications

I definitely do not see this option, I can only enable or disable audible notifications, there is no option to set a local sound file.

**Comment by t3chguy** [comment]
<img width="781" height="575" alt="Image" src="https://github.com/user-attachments/assets/27b5d0b9-425d-44a9-ae80-bcc72f356c2b" />

**Comment by rkrisztian** [comment]
@t3chguy, I find this confusing. Why is it called notification sound instead of ringing sound or ringtone? Maybe we're talking about a different feature? Notification sounds seem to be about for receiving messages, not getting a call.

**Comment by t3chguy** [comment]
Oh sorry it is just the notification sound not the ringing sound. The title of this issue refers to both.

**Comment by mrkesu** [comment]
I definitely do not see this in Element version 1.12.10 running on Windows 11 (and just to make sure I hit "check for update" and this is the newest version).

Maybe the CO2 level in my room is dangerously high because your left hand menu and notification options don't even look like mine does 🥴

<img width="953" height="571" alt="Image" src="https://github.com/user-attachments/assets/01e02f78-57b2-4f3d-bf05-7f6e7f5c7f8d" />

**Comment by t3chguy** [comment]
@mrkesu I suggest reading my message more clearly.

<img width="957" height="203" alt="Image" src="https://github.com/user-attachments/assets/7f010787-3b28-476c-a951-c8ab7271ae85" />

**Comment by KaKi87** [comment]
In other words, you have to change the notification sound for every room you're a member of, and you still don't get to change other sounds.

**Comment by PietG1** [comment]
It should be handy if there is an indication of successful upload/save and the save button stays enabled and can bev clicked multiple times to send the upload post
